essence-of-live-coding/src/LiveCoding/RuntimeIO/Launch.hs

```diff
@@ -32,8 +35,11 @@ class Monad m => Launchable m where
 instance Launchable IO where
   runIO = id
 
-instance Launchable (StateT (HandlingState IO) IO) where
-  runIO = runHandlingState
+instance (Typeable m, Launchable m) => Launchable (StateT (HandlingState m) m) where
+  runIO = runIO . runHandlingState
+
+instance (Data e, Finite e, Launchable m) => Launchable (ExceptT e m) where
+  runIO liveProgram = runIO $ foreverCLiveProgram $ tryLiveProgram liveProgram
```

Comment:
Maybe it makes more sense to stop the program upon exceptions. Looping forever is something the user can do themselves before they pass it to `runIO`.

Reply:
But this is not possible with the current definition of `Launchable`, since it translates to a `LiveProgram IO`, which can never terminate.